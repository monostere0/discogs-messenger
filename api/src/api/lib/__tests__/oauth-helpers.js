const {
  getAppHeaders,
  getSecureHeaders,
  getInitialHeaders,
  getAuthorizationHeaders,
} = require('../oauth-helpers');


jest.mock('../../../../conf', () => function mockConf() {
  return {
    app_user_agent: 'FooAgent/1.0',
    discogs: {
      oauth: {
        callback_url: 'http://callback-url',
      },
    },
  };
});
jest.mock('../math', () => ({
  round(x) { return x; },
  random() {
    return 0.123456789;
  },
}));

describe('app/lib/oauthHelpers', () => {
  global.Date = class Date {
    getTime() {
      return 13234523452;
    }
  };

  describe('getAppHeaders', () => {
    it('should return the app headers', () => {
      expect(getAppHeaders('sampleHeaders')).toMatchSnapshot();
    });
  });

  describe('getSecureHeaders', () => {
    it('should return the app oAuth headers', () => {
      expect(getSecureHeaders({
        oauth_token: 'oAuthTokenValue',
        oauth_token_secret: 'oAuthSecretValue',
      })).toMatchSnapshot();
    });
  });

  describe('getInitialHeaders', () => {
    it('should return the initial call headers', () => {
      expect(getInitialHeaders()).toMatchSnapshot();
    });
  });

  describe('getAuthorizationHeaders', () => {
    it('should throw an error if the stored secret is null', () => {
      expect(() => getAuthorizationHeaders('foo', 'bar')).toThrow();
    });

    it('should return the authorization headers if the param is supplied', () => {
      expect(getAuthorizationHeaders('foo', 'bar', 'baz')).toMatchSnapshot();
    });
  });
});
