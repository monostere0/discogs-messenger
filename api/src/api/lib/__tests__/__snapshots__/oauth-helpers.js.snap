// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/lib/oauthHelpers getAppHeaders should return the app headers 1`] = `
Object {
  "Accept": "application/vnd.discogs.v2.plaintext+json",
  "Authorization": "sampleHeaders",
  "Content-Type": "application/x-www-form-urlencoded",
  "User-Agent": "FooAgent/1.0",
}
`;

exports[`app/lib/oauthHelpers getAuthorizationHeaders should return the authorization headers if the param is supplied 1`] = `"OAuth oauth_consumer_key=\\"undefined\\", oauth_nonce=\\"616335b9.5440ec\\", oauth_token=foo, oauth_signature=\\"undefined%26baz\\", oauth_signature_method=\\"PLAINTEXT\\", oauth_timestamp=\\"13234523452\\", oauth_verifier=\\"bar\\""`;

exports[`app/lib/oauthHelpers getInitialHeaders should return the initial call headers 1`] = `"OAuth oauth_consumer_key=\\"undefined\\", oauth_nonce=\\"616335b9.5440ec\\", oauth_signature=\\"undefined&\\", oauth_signature_method=\\"PLAINTEXT\\", oauth_timestamp=\\"13234523452\\", oauth_callback=\\"undefined\\""`;

exports[`app/lib/oauthHelpers getSecureHeaders should return the app oAuth headers 1`] = `"OAuth oauth_consumer_key=\\"undefined\\", oauth_nonce=\\"616335b9.5440ec\\", oauth_signature=\\"undefined%26oAuthSecretValue\\", oauth_signature_method=\\"PLAINTEXT\\", oauth_timestamp=\\"13234523452\\", oauth_token=\\"oAuthTokenValue\\""`;
